.github/workflows/codeql-buildscript.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```

Comment:
Can the job use `tests/compile-all/build.sh` for the functionality of this script?

(That script should also stop running "make -j" without any upper limit of parallelism, but that isn't your problem.)

Reply:
And just be aware that the `tests/compile-all/build.sh` takes a while, output from `time -p make` in that directory:
```
real 2043.81
user 13620.13
sys 6830.35
```

Reply:
Yes, the job can be subbed out with any script.

Reply:
You can replace this script with a call to `make -j<cores> -C tests/compile-all` now that #2747 is merged.

If you manage to run this job inside the Docker container, we have an alias called `cimake` so you can get an even better ccache hit rate by running `cimake -j<cores> -C tests/compile-all` instead.